lib/recaptcha/client_helper.rb

```diff
@@ -84,5 +85,52 @@ def self.recaptcha_components(options = {})
 
       [html, tag_attributes, fallback_uri]
     end
+
+    private
+
+    def default_callback_name
+      'invisibleRecaptchaSubmit'
+    end
+
+    def default_callback_script
+      <<-HTML
+        <script>
+          var createClosestPolyfill = function () {
+            if (!Element.prototype.matches) {
+              Element.prototype.matches =
+                Element.prototype.msMatchesSelector ||
+                Element.prototype.webkitMatchesSelector;
+            }
+            Element.prototype.closest = function(s) {
+                var el = this;
+                var ancestor = this;
+                if (!document.documentElement.contains(el)) { return null };
+                do {
+                    if (ancestor.matches(s)) return ancestor;
+                    ancestor = ancestor.parentElement;
+                } while (ancestor !== null);
+                return null;
+            };
+          };
+          var #{default_callback_name} = function () {
+            var ele = document.getElementsByClassName('g-recaptcha');
+            if (ele.length > 0) {
+              if (!Element.prototype.closest) {
```

Comment:
do something simpler like iterating parents until one is a form ?

Reply:
👍